Various code view improvements

silverwind opened the pull request:
1. Restore missing styles for message close icon
2. Move `code-line-button` so that it does not go off-screen on small viewports
3. Make `code-line-button` look and behave like other buttons
4. Make `code-line-button` work in blame
5. Make the active selection span the whole line, not just the code part
6. Tweak colors, make dark theme code bg darker, make line numbers same color in diff and file view.
7. Move code background to parent, fixing border radius and other problems
8. Enable code wrap in blame
9. Improve blame responsiveness
10. Remove `--color-code-sidebar-bg` in blame, now it uses same background as code
11. Rename `--color-active-line` to `--color-highlight-bg`
12. Add `--color-highlight-bg`
13. Fix button group borders on hover and border-right on last button.

<img width="1343" alt="Screenshot 2024-03-23 at 22 34 13" src="https://github.com/go-gitea/gitea/assets/115237/fcbb919f-5dc3-43f0-97f6-870d6f412554">
<img width="1334" alt="Screenshot 2024-03-23 at 22 34 26" src="https://github.com/go-gitea/gitea/assets/115237/ca44c3b7-4328-4645-ba49-b0dc6a5ac06d">

<img width="1338" alt="Screenshot 2024-03-23 at 22 34 57" src="https://github.com/go-gitea/gitea/assets/115237/00eb0b5a-1ec7-4669-a94a-4602b9d1c1ac">
<img width="1337" alt="Screenshot 2024-03-23 at 22 34 42" src="https://github.com/go-gitea/gitea/assets/115237/752edc4a-064f-413c-9dff-c086187fcd85">

Fixes: https://github.com/go-gitea/gitea/issues/18074

yardenshoham commented:
Did... Did you just add jQuery code to the codebase?

yardenshoham commented:
What have I done to deserve this

delvh commented:
I think the problem is that this file is already using jQuery heavily, so using something else would only be more confusing at the moment.

yardenshoham commented:
Fine :(

silverwind commented:
Can rewrite without I guess. The whole function was very hard to follow because of those confusing variable names it has.

silverwind commented:
I guess this is as much refactor as I'm willing to do right now. Variable names are much clearer now and some jQuery is removed.

silverwind commented:
More tweaks done. Code sidebar is gone on blame. Two new CSS variables for the highlight added. Blame button muted, background moved to parent, fixing border radius issue.

<img width="1309" alt="Screenshot 2024-03-23 at 15 48 00" src="https://github.com/go-gitea/gitea/assets/115237/ce77988d-ec8a-4cf9-a128-8f491898bc76">
<img width="1306" alt="Screenshot 2024-03-23 at 15 48 14" src="https://github.com/go-gitea/gitea/assets/115237/c4a4715f-d920-4c72-897b-0922e40577c7">

<img width="1315" alt="Screenshot 2024-03-23 at 15 48 46" src="https://github.com/go-gitea/gitea/assets/115237/3f0f3dcd-db8f-4b41-8db1-8973832bdb22">
<img width="1315" alt="Screenshot 2024-03-23 at 15 48 28" src="https://github.com/go-gitea/gitea/assets/115237/4535999f-c902-48f5-ab8d-d5075a6a89a2">

silverwind commented:
Unescape button was missing right border in blame view, fixed in https://github.com/go-gitea/gitea/pull/30014/commits/b867329e52568770796d5d7fc2c6ed7666c9b55d so that 2nd and 2nd to last button is considered for border addition:

<img width="158" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/f1db7a1b-05c6-46de-8508-b8bdc87e2950">

silverwind commented:
Blame commit links now look like this:

<img width="168" alt="Screenshot 2024-03-23 at 16 09 39" src="https://github.com/go-gitea/gitea/assets/115237/177eb88c-4c66-40f0-b92e-3f7b1b733568">

I also added underline styles to base, I think they were lost as a regression from https://github.com/go-gitea/gitea/pull/29980.

silverwind commented:
Regression with `a:hover` from https://github.com/go-gitea/gitea/pull/29980 confirmed. I cleaned up the styles and also added a devtest example for links:

<img width="271" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/34334bbf-cf55-45c0-8f48-69d46b3c8acf">

silverwind commented:
Screenshots updated in OP. Ready to go imho.

silverwind commented:
Some more minor fixed in blame:

<img width="1332" alt="Screenshot 2024-03-23 at 22 58 41" src="https://github.com/go-gitea/gitea/assets/115237/2bef78c5-6d49-4340-989d-b7509998cafa">

silverwind commented:
I also enabled code wrapping in blame view now:

<img width="1327" alt="Screenshot 2024-03-23 at 23 41 09" src="https://github.com/go-gitea/gitea/assets/115237/a8b26802-82fc-4da1-8b70-0dd48e4c9204">

silverwind commented on web_src/css/modules/button.css:
```diff
@@ -61,11 +61,17 @@ It needs some tricks to tweak the left/right borders with active state */
+.ui.buttons .button:nth-last-child(2):has(+ .button.gt-hidden) {
   border-right: 1px solid var(--color-light-border);
 }
 
```
Actually, I found a fix so this approximation is no longer needed by just styling all hover styles explicitely with an opaque color.

6543 commented on web_src/js/features/repo-code.js:
```diff
@@ -47,17 +53,15 @@ function selectRange($list, $select, $from) {
-    if ($copyPermalink.length === 0) {
-      return;
-    }
+    if ($copyPermalink.length === 0) return;
```
```suggestion
    if (!$copyPermalink.length) return;
```

too?

silverwind commented:
I will try to get `code-line-button` not to hide line number is between 1 and 3 digits, e.g. like GitHub:

<img width="102" alt="image" src="https://github.com/go-gitea/gitea/assets/115237/30ebe956-6f50-44f7-ac3e-be98fc640f1c">

6543 reviewed (approved):
lets merge this and itterate on it
